### DISCUSSION TITLE
material derivative w.r.t gradvar

### DISCUSSION THREAD

**Discussion by anupkgeos** [idaholab/moose#26139] [Q&A General]
Hi 
I want to implement porosity change as a function of strain:

$$ \phi = 1 - (1-\phi_0) \exp^{-\epsilon} $$

```
(*_dporosity_dvar)[_qp][v] = (1-phi0)*std::exp(-vol_strain) 

(*_dporosity_dgradvar)[_qp][v]  = how?
```
how to calculate the derivative with respect gradient?

**Comment by GiudGiud** [comment]
Hello

Do you mean in terms of math or moose?
For the math I d Taylor expand the expression then that makes the gradient appear and you can take the derivative there

For MOOSE you can retrieve derivative of material properties the same way as you retrieve material properties. You just need to have defined that derivative material property somewhere else

Guillaume
